unified_planning/model/mixins/user_types_set.py

```diff
@@ -41,7 +41,7 @@ def _add_user_type(self, type: "up.model.types.Type"):
         assert type.is_user_type()
         if type not in self._user_types:
             t = cast(_UserType, type)
-            if self._has_name_method(t.name):
+            if self.has_type(t.name):
```

Comment:
Same as above, why do we need this?

Reply:
Same (with a different domain requiring this one).

Reply:
Handled together with the previous issue.